google-cloud-storage/src/main/scala/akka/stream/alpakka/googlecloud/storage/GoogleCloudStorageModel.scala

```diff
@@ -0,0 +1,389 @@
+
+    /** Java API */
+    def withContentType(value: akka.http.javadsl.model.ContentType): StorageObject =
+      copy(contentType = value.asInstanceOf[ContentType])
+    def withSize(value: Long): StorageObject = copy(size = value)
+    def withEtag(value: String): StorageObject = copy(etag = value)
+    def withMd5Hash(value: String): StorageObject = copy(md5Hash = value)
+    def withCrc32c(value: String): StorageObject = copy(crc32c = value)
+    def withMediaLink(value: String): StorageObject = copy(mediaLink = value)
+
+    private def copy(
+        kind: String = kind,
+        id: String = id,
+        name: String = name,
+        bucket: String = bucket,
+        generation: Long = generation,
+        contentType: ContentType = contentType,
+        size: Long = size,
+        etag: String = etag,
+        md5Hash: String = md5Hash,
+        crc32c: String = crc32c,
+        mediaLink: String = mediaLink
+    ): StorageObject = new StorageObject(
+      kind = kind,
+      id = id,
+      name = name,
+      bucket = bucket,
+      generation = generation,
+      contentType = contentType,
+      size = size,
+      etag = etag,
+      md5Hash = md5Hash,
+      crc32c = crc32c,
+      mediaLink = mediaLink
+    )
+
+    override def toString =
+      "StorageObject(" +
+      s"kind=$kind," +
+      s"id=$id," +
+      s"name=$name," +
+      s"bucket=$bucket," +
+      s"generation=$generation," +
+      s"contentType=$contentType," +
+      s"size=$size," +
+      s"etag=$etag," +
+      s"md5Hash=$md5Hash," +
+      s"crc32c=$crc32c," +
+      s"mediaLink=$mediaLink" +
+      ")"
+
+    override def equals(other: Any): Boolean = other match {
+      case that: StorageObject =>
+        java.util.Objects.equals(this.kind, that.kind) &&
+        java.util.Objects.equals(this.id, that.id) &&
+        java.util.Objects.equals(this.name, that.name) &&
+        java.util.Objects.equals(this.bucket, that.bucket) &&
+        java.util.Objects.equals(this.generation, that.generation) &&
+        java.util.Objects.equals(this.contentType, that.contentType) &&
+        java.util.Objects.equals(this.size, that.size) &&
+        java.util.Objects.equals(this.etag, that.etag) &&
+        java.util.Objects.equals(this.md5Hash, that.md5Hash) &&
+        java.util.Objects.equals(this.crc32c, that.crc32c) &&
+        java.util.Objects.equals(this.mediaLink, that.mediaLink)
+      case _ => false
+    }
+
+    override def hashCode(): Int =
+      java.util.Objects.hash(kind,
+                             id,
+                             name,
+                             bucket,
+                             Long.box(generation),
+                             contentType,
+                             Long.box(size),
+                             etag,
+                             md5Hash,
+                             crc32c,
+                             mediaLink)
+  }
+
+  object StorageObject {
+
+    /** Scala API */
+    def apply(
```

Comment:
It not really useful to offer factory methods taking all fields (we hide the constructor so nobody can depend on the API listing all fields). 
Just offer factory methods with the important values, others should be added with the `withXyz` methods.